inspirehep/modules/workflows/tasks/refextract.py

```diff
@@ -184,3 +189,46 @@ def extract_references_from_raw_ref(reference, custom_kbs_file=None):
     return extract_references_from_text(
         raw_ref['value'], source=raw_ref['source'], custom_kbs_file=custom_kbs_file
     )
+
+
+def match_references(references):
+    """Match references to record ids given a list of references.
+
+    Args:
+        references: Schema compliant list of references
+
+    Returns:
+        associated_references: The same list with the associated record ids (if any)
+    """
+
+    for i, reference in enumerate(references):
+        matched_recid = match_reference(reference)
+        if matched_recid:
+            references[i]['record'] = get_record_ref(matched_recid, 'literature')
```

Comment:
I would drop this function, and have a `match_reference` that does all the work for one reference, that is take a reference and spit out a reference, potentially with the `record` key set to the matched record.
You would then replace calls of `result = match_references(references)` by `result = [match_reference(ref) for ref in references]`.